api/src/main/java/org/pmiops/workbench/reporting/ReportingUploadServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package org.pmiops.workbench.reporting;
+
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.model.ReportingResearcher;
+import org.pmiops.workbench.model.ReportingSnapshot;
+import org.pmiops.workbench.model.ReportingWorkspace;
+import org.pmiops.workbench.reporting.insertion.ReportingInsertionJobBuilder;
+import org.pmiops.workbench.reporting.insertion.ResearcherParameter;
+import org.pmiops.workbench.reporting.insertion.WorkspaceParameter;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Service;
+
+@Service
+@Primary
+public class ReportingUploadServiceImpl implements ReportingUploadService {
+  private static final Logger logger = Logger.getLogger("ReportingUploadServiceInsertQueryImpl");
+  private static final long MAX_WAIT_TIME = 10000;
+
+  private final BigQueryService bigQueryService;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  private static final ReportingInsertionJobBuilder<ReportingResearcher> researcherJobBuilder =
+      ResearcherParameter::values;
+  private static final ReportingInsertionJobBuilder<ReportingWorkspace> workspaceJobBuilder =
+      WorkspaceParameter::values;
+
+  public ReportingUploadServiceImpl(
+      BigQueryService bigQueryService, Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.bigQueryService = bigQueryService;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+  }
+
+  @Override
+  public ReportingJobResult uploadSnapshot(ReportingSnapshot reportingSnapshot) {
```

Comment:
Exactly. We negotiated a schema with the analytics team, who wish to access this dataset from tools like Tableau and NyFy, and we're deliberately keeping things very simple on the representation side. Even though it's effectively a DB dump every day, we don't expect the overall performance will suffer much.

There's also  BigQuery [struct](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#declaring_a_struct_type) data  types. If we  were going for a nested representation, I'd probably use that before JSON  here so that we have a concrete schema.